brain/python/synapses.cpp

```diff
@@ -0,0 +1,172 @@
+    out.append( v.y( ));
+    out.append( v.z( ));
+    return out;
+}
+
+#define GET_SYNAPSE_VEC3_PROPERTY( name ) \
+    bp::list Synapse_##name( const SynapseWrapper& synapse )    \
+        { return _toList( synapse.name( )); }
+
+GET_SYNAPSE_VEC3_PROPERTY( getPresynapticSurfacePosition )
+GET_SYNAPSE_VEC3_PROPERTY( getPresynapticCenterPosition )
+GET_SYNAPSE_VEC3_PROPERTY( getPostsynapticSurfacePosition )
+GET_SYNAPSE_VEC3_PROPERTY( getPostsynapticCenterPosition )
+
+}
+
+void export_Synapses()
+{
+
+bp::class_< SynapseWrapper >( "Synapse", bp::no_init )
+    .def( "pre_gid", &Synapse::getPresynapticGID )
+    .def( "pre_section", &Synapse::getPresynapticSectionID )
+    .def( "pre_segment", &Synapse::getPresynapticSegmentID )
+    .def( "pre_distance", &Synapse::getPresynapticDistance )
+    .def( "pre_center_position", Synapse_getPresynapticCenterPosition )
+    .def( "pre_surface_position", Synapse_getPresynapticSurfacePosition )
+    .def( "post_gid", &Synapse::getPostsynapticGID )
+    .def( "post_section", &Synapse::getPostsynapticSectionID )
+    .def( "post_segment", &Synapse::getPostsynapticSegmentID )
+    .def( "post_distance", &Synapse::getPostsynapticDistance )
+    .def( "post_center_position", Synapse_getPostsynapticCenterPosition )
+    .def( "post_surface_position", Synapse_getPostsynapticSurfacePosition )
+    .def( "delay", &Synapse::getDelay )
+    .def( "conductance", &Synapse::getConductance )
+    .def( "utilization", &Synapse::getUtilization )
+    .def( "depression", &Synapse::getDepression )
+    .def( "facilitation", &Synapse::getFacilitation )
+    .def( "decay", &Synapse::getDecay )
+    .def( "efficacy", &Synapse::getEfficacy )
+    ;
+
+bp::class_< SynapsesWrapper >( "Synapses", bp::no_init )
+    .def( "__nonzero__", nonzero )
+    .def( "__len__", &Synapses::size )
+    .def( "__getitem__", Synapses_get )
+    // There is no need to wrap the iterator, Python provides one out of the
+    // box thanks to __len__ and __getitem__
+    .def( "empty", &Synapses::empty )
+    .def( "indices", Synapses_indices )
+    .def( "pre_gids", Synapses_preGIDs )
+    .def( "pre_section_ids", Synapses_preSectionIDs )
+    .def( "pre_segment_ids", Synapses_preSegmentIDs )
+    .def( "pre_distances", Synapses_preDistances )
+    .def( "pre_surface_x_positions", Synapses_preSurfaceXPositions )
+    .def( "pre_surface_y_positions", Synapses_preSurfaceYPositions )
+    .def( "pre_surface_z_positions", Synapses_preSurfaceZPositions )
+    .def( "pre_center_x_positions", Synapses_preCenterXPositions )
+    .def( "pre_center_y_positions", Synapses_preCenterYPositions )
+    .def( "pre_center_z_positions", Synapses_preCenterZPositions )
+    .def( "post_gids", Synapses_postGIDs )
+    .def( "post_section_ids", Synapses_postSectionIDs )
+    .def( "post_segment_ids", Synapses_postSegmentIDs )
+    .def( "post_distances", Synapses_postDistances )
+    .def( "post_surface_x_positions", Synapses_postSurfaceXPositions )
```

Comment:
Tuples are probably better than lists. I thought about numpy arrays, but they have to be created in the C++ side and the code I have already in place is not sufficient. If the client needs them you can always do it easily in Python.
If we see that it's better to return arrays we can change it later without breaking code relying on the list/tuple interface.